### PR TITLE
fix: datepicker doesnt close on first selection of date on month change

### PR DESCRIPTION
**Pull request by SaiCharanChetpelly31**
Fixes issue #30701 

## Summary by CodeRabbit

- **New Features**
  - Introduced new tests for the DatePicker widget to ensure proper functionality and behavior when dates are selected and months are changed.

- **Bug Fixes**
  - Improved date selection handling to close the datepicker on a single click when the month is changed.

### CODE REVIEW

**Comment by SaiCharanChetpelly31** [comment]
@ApekshaBhosale I accidentally deleted my forked repository, but I have raised PR with the suggested changes addressing the comments. Could you please review this PR?

**Comment by SaiCharanChetpelly31** [comment]
Hi @ApekshaBhosale 

I hope you're well. I’m curious and excited to hear your thoughts on my pull request raised a while ago. Looking forward to your review!
Thank you.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Datepicker/DatePicker4_spec.js
Comment (jsartisan) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Datepicker/DatePicker4_spec.js:1]: Hey @SaiCharanChetpelly31 Can you create this in typescript? We have a check in github that we don't allow new tests in javascript. We would need it in typescript.

Comment (SaiCharanChetpelly31) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Datepicker/DatePicker4_spec.js:1]: @jsartisan I updated spec to ts. Can you please review?

**Comment by SaiCharanChetpelly31** [comment]
Hi @jsartisan Gentle Remainder: I have addressed review comment, can you please check ?
Thanks in advance.

**Review by ramsaptami — APPROVED** [review]

**Comment by SaiCharanChetpelly31** [comment]
Hi @ramsaptami Could you please merge this PR if everything is ok?

**Comment by ramsaptami** [comment]
@jsartisan please review and merge if everything is ok

**Comment by sagar-qa007** [comment]
@jsartisan  Please check this PR.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Datepicker/DatePicker4_spec.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    });
+
+    const navigateAndSelectRandomDate = (direction: string) => {
+      const buttonClass =
+        direction === "prev"
+          ? ".DayPicker-NavButton--prev"
+          : ".DayPicker-NavButton--next";
+      agHelper.AssertElementVisibility(buttonClass, true, 0);
+      agHelper.GetNClick(buttonClass);
+      cy.get(".DayPicker-Month")
```
Comment (sagar-qa007) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Datepicker/DatePicker4_spec.ts:24]: Please use the locator files for storing any locators. Also add the data-testid for locators.

**Review by sagar-qa007 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**